src/wallet.rs

```diff
@@ -670,3 +716,66 @@ impl FromStr for Chain {
 #[derive(Debug, thiserror::Error)]
 #[error("Unsupported chain: {0}")]
 pub struct WrongChain(String);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct EncryptedWallet {
+    name: String,
+    encrypted_xprv: Vec<u8>,
+    sk_salt: [u8; 32],
+    chain: Chain,
+}
+
+impl EncryptedWallet {
+    pub fn decrypt(self, encryption_key: impl AsRef<[u8]>) -> Result<Wallet> {
```

Comment:
Is `encryption_key` just the password that the user supplies?

In other words, is this safe to use with a user-chosen password or should the calling code do some more key generation?

Some docs (with examples perhaps?) would be nice :)

Reply:
> Is `encryption_key` just the password that the user supplies?
> 
> In other words, is this safe to use with a user-chosen password or should the calling code do some more key generation?

I think using a password here fine as the encryption key is the password to unlock the wallet.

Was not sure what to call it because it is an encryption key but it functions like a password.

Reply:
I usually see it as:

- `key` for me implies that it must have sufficient entropy etc to act as direct input to an encryption algorithm like AES.
- `password` is something the user chooses and remembers

You go from password to key using KDF.

We are already using a KDF here unless that has been changed so I would call this a `password` to let API users know that they don't need to run a KDF first before calling this function.